src/MemcachedCachePool.php

```diff
@@ -53,12 +62,15 @@ protected function clearOneObjectFromCache($key)
         return $this->cache->getResultCode() === \Memcached::RES_NOTFOUND;
     }
 
-    protected function storeItemInCache($key, CacheItemInterface $item, $ttl)
+    protected function storeItemInCache($key, CacheItemInterface $item, $ttl = -1)
     {
-        if ($ttl === null) {
-            $ttl = 0;
-        }
+        $key = $this->getHierarchyKey($key);
 
-        return $this->cache->set($key, $item, $ttl);
+        return $this->cache->set($key, $item, 0, $ttl);
```

Comment:
I think it slipped in an error here. Memcached::set() has only three parameters. 

Don't we have to make sure `$ttl` is not null?